### PR TITLE
NIAD-24: enhancement: produce a postman collection of requests that can be made to adaptors

### PR DESCRIPTION
**Pull request by Alicja-Kainos**
Request for forward-reliable pattern to be added

### CODE REVIEW

**Review by grzegorz-robak — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -143,10 +143,12 @@ mhs-reference-implementation/selenium_tests/data/local_asid
 /mhs/spineroutelookup/nhs-spineroutelookup-env.yaml
 export-env-vars-and-run-mhs-docker.sh
 /integration-tests/integration_tests/integration-test-env.yaml
-/integration-tests/integration_tests/integration-tests-env.yaml
 
 # system
 .DS_Store
 
+# enviornment file
```
Comment (bartek-sarul) [inline .gitignore:150]: Why this change? All were together under line 140 category

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,3 +1,23 @@
 {
-    "python.pythonPath": "/usr/local/bin/python3"
-}
\ No newline at end of file
+    "python.pythonPath": "/usr/local/bin/python3",
+    "workbench.settings.editor": "json",
+    "workbench.settings.useSplitJSON": true,
+    "rest-client.environmentVariables": {
```
Comment (bartek-sarul) [inline .vscode/settings.json:5]: What will happen if someone adds new configuration? If this file be will be affected then I guess it shouldn't be versioned

File: rest-client/README.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+- VSCode Extension Rest Client: https://marketplace.visualstudio.com/items?itemName=humao.rest-client 
+
+- VSCode Extension GUID Insert: https://marketplace.visualstudio.com/items?itemName=heaths.vscode-guid
+
+Open workspace, ensure Rest Client and GUID Insert are enabled on workspace. Go to `Preferences` > `Extensions`, search for the two extensions above (Rest Client & GUID) and select enable. 
+
+</br>
+
+## Constructing Request
```
Comment (bartek-sarul) [inline rest-client/README.md:17]: To be honest I'm not sure why the below is here. Isn't this a kind of copy paste from official documentation? Instead of listing this here I prefer adding a link to documentation.
Leave only custom stuff or tips

File: rest-client/mhs/healthchecks/Healthcheck.http [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Health check for MHS
+
+# @name Healthcheck
+GET {{HOSTNAME}}/healthcheck HTTP/1.1
```
Comment (bartek-sarul) [inline rest-client/mhs/healthchecks/Healthcheck.http:4]: Each service has healthcheck so if you deploy all, those will be on different ports

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# @name Async_Express_Pattern_Message
+
+POST {{HOSTNAME}}:{{OUTBOUND-PORT}} HTTP/1.1
+content-type: application/json
```
Comment (bartek-sarul) [inline rest-client/mhs/outbound/Async-Express-Pattern-Message.http:4]: It probably doesn't matter much, but the convention of HTTP standard is that Content-Type is camel case with a dash separator

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+Interaction-Id: QUPC_IN160101UK05
+sync-async: true
+# This interaction allows both sync and async response. If false is, than acknowledgemnt of message
```
Comment (bartek-sarul) [inline rest-client/mhs/outbound/Async-Express-Pattern-Message.http:7]: `If false is, ` ?

File: rest-client/mhs/spine/async_express.http [hunk 7]
```diff
@@ -0,0 +1,101 @@
+
+# @name Async_Express
+POST https://192.168.128.11/reliablemessaging/queryrequest HTTP/1.1
```
Comment (bartek-sarul) [inline rest-client/mhs/spine/async_express.http:3]: IP ?

File: rest-client/mhs/inbound/Post.http [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (bartek-sarul) [inline rest-client/mhs/inbound/Post.http:1]: What's the purpose of this empty request?

File: rest-client/messages/QUPC_IN160101UK05.json [hunk 9]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (bartek-sarul) [inline rest-client/messages/QUPC_IN160101UK05.json:1]: What's the purpose of this file?

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
...         </author1>\r\n                <query>\r\n                    <dissentOverride>\r\n                        <semanticsText>DissentOverride</semanticsText>\r\n                        <value code=\"0\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.17.60\" displayName=\"Demonstration\">\r\n                            <originalText>Demonstration</originalText>\r\n                        </value>\r\n                    </dissentOverride>\r\n                    <filterParameters>\r\n                        <date>\r\n                            <semanticsText>Date</semanticsText>\r\n                            <value>\r\n                                <low value=\"20100908161126\"/>\r\n                                <high value=\"20190927152034\"/>\r\n                            </value>\r\n                        </date>\r\n                    </filterParameters>\r\n                    <nHSNumber>\r\n                        <semanticsText>NHSNumber</semanticsText>\r\n                        <value root=\"2.16.840.1.113883.2.1.4.1\" extension=\"9689177923\" />\r\n                    </nHSNumber>\r\n                </query>\r\n            </ControlActEvent>\r\n</QUPC_IN160101UK05>\r\n"
```
Comment (bartek-sarul) [inline rest-client/mhs/outbound/Async-Express-Pattern-Message.http:12]: Is it possible to pretty print the xml or import it from another file?

File: rest-client/mhs/healthchecks/Healthcheck.http (hunk 4 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/healthchecks/Healthcheck.http:4]: I remove this healthcheck, there are other healthchecks, one er each service

File: .vscode/settings.json (hunk 2 above)
Comment (Alicja-Kainos) [reply .vscode/settings.json:5]: There is .vscode in .gitignore file, I am not sure why this is versioned

Comment (grzegorz-robak) [reply .vscode/settings.json:5]: I believe it should be `.vscode/` (notice the slash) in .gitignore

Comment (Alicja-Kainos) [reply .vscode/settings.json:5]: All right, @grzegorz-robak , let's try it :)

File: rest-client/README.md (hunk 3 above)
Comment (Alicja-Kainos) [reply rest-client/README.md:17]: Link added instead of this

File: rest-client/messages/QUPC_IN160101UK05.json (hunk 9 above)
Comment (Alicja-Kainos) [reply rest-client/messages/QUPC_IN160101UK05.json:1]: I think it could be removed.

File: rest-client/mhs/inbound/Post.http (hunk 8 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/inbound/Post.http:1]: It has been copied from Postman, where also was empty. Should we just remove the request for the inbound service? I think that the inbound service is tested together with outbound when requesting outbound. Am I right?

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http (hunk 10 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/outbound/Async-Express-Pattern-Message.http:12]: I'll see what can be done, if it is possible to pass the file content as a string instead of this.

File: rest-client/mhs/inbound/Post.http (hunk 8 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/inbound/Post.http:1]: The same question about the spine - it is also tested together with outbound.

File: rest-client/mhs/spine/async_express.http (hunk 7 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/spine/async_express.http:3]: It is the hostname of spine, @Broom94 , isn't it? I've just replace it with configurable parameter.

Comment (grzegorz-robak) [reply rest-client/mhs/spine/async_express.http:3]: OpenTest doesn't work with hostnames, that's why we use IP

File: .vscode/settings.json (hunk 2 above)
Comment (Alicja-Kainos) [reply .vscode/settings.json:5]: it works

File: rest-client/mhs/spine/async_express.http (hunk 7 above)
Comment (Alicja-Kainos) [reply rest-client/mhs/spine/async_express.http:3]: so should I add this IP as a parameter or leave it like this in the request?

Comment (grzegorz-robak) [reply rest-client/mhs/spine/async_express.http:3]: I'd leave it as is. It's kind of constant and when hostname will work we can just update this line. What do you think @bartek-sarul ?

File: rest-client/mhs/inbound/Post.http (hunk 8 above)
Comment (Broom94) [reply rest-client/mhs/inbound/Post.http:1]: inbound requests now added for unsolicited and 'solicited' if thats the correct term

File: rest-client/mhs/outbound/Async-Express-Pattern-Message.http (hunk 10 above)
Comment (Broom94) [reply rest-client/mhs/outbound/Async-Express-Pattern-Message.http:12]: We can add files but these files cannot contain environment or file variables which these request use. Also we cannot have it in pretty print as we get invalid format error, so whole body needs to be on one line.
